staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch_test.go

```diff
@@ -271,19 +276,27 @@ testCases:
 `)
 
 func TestSortMergeLists(t *testing.T) {
+	mergeItemOpenapiSchema := PatchMetaFromOpenAPI{
+		Schema: sptest.GetMergeItemSchemaOrDie(),
+	}
+	schemas := []LookupPatchMeta{mergeItemStructSchema, mergeItemOpenapiSchema}
+
 	tc := SortMergeListTestCases{}
 	err := yaml.Unmarshal(sortMergeListTestCaseData, &tc)
 	if err != nil {
 		t.Errorf("can't unmarshal test cases: %s\n", err)
 		return
 	}
 
-	for _, c := range tc.TestCases {
-		got := sortJsonOrFail(t, testObjectToJSONOrFail(t, c.Original), c.Description)
-		expected := testObjectToJSONOrFail(t, c.Sorted)
-		if !reflect.DeepEqual(got, expected) {
-			t.Errorf("error in test case: %s\ncannot sort object:\n%s\nexpected:\n%s\ngot:\n%s\n",
-				c.Description, mergepatch.ToYAMLOrError(c.Original), mergepatch.ToYAMLOrError(c.Sorted), jsonToYAMLOrError(got))
+	for _, schema := range schemas {
```

Comment:
If that test fails, how do I know from the error message which variant of the schema failed?

Reply:
Good point. It worth pointing out.